### PR TITLE
[mso_schema_site_anp_epg_staticport] Made changes to reduce Schema API request size by using list-identity

### PR DESCRIPTION
**Pull request by anvitha-jain**
… respect to schema usage

### CODE REVIEW

File: plugins/module_utils/mso.py [hunk 1]
```diff
@@ -947,3 +947,10 @@ def update_filter_obj(self, contract_obj, filter_obj, filter_type, contract_disp
         filter_obj['contractScope'] = contract_obj.get('scope')
         filter_obj['contractFilterType'] = contract_obj.get('filterType')
         return filter_obj
+
+    def get_schema_details(self, schema):
+        schema_summary = self.query_objs('schemas/list-identity', key='schemas', displayName=schema)
+        if not schema_summary:
+            self.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
+        schema_id = schema_summary[0].get('id')
```
Comment (lhercot) [inline plugins/module_utils/mso.py:955]: What if query_objs return more than one result? Is that possible?

File: plugins/module_utils/mso.py [hunk 2]
```diff
@@ -947,3 +947,10 @@ def update_filter_obj(self, contract_obj, filter_obj, filter_type, contract_disp
         filter_obj['contractScope'] = contract_obj.get('scope')
         filter_obj['contractFilterType'] = contract_obj.get('filterType')
         return filter_obj
+
+    def get_schema_details(self, schema):
```
Comment (lhercot) [inline plugins/module_utils/mso.py:951]: Is it the details or just the id? Name is probably not correct

File: plugins/modules/mso_schema_site_anp_epg_staticport.py [hunk 3]
```diff
@@ -284,11 +284,9 @@ def main():
     mso = MSOModule(module)
 
     # Get schema_id
-    schema_obj = mso.get_obj('schemas', displayName=schema)
-    if not schema_obj:
-        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))
-    schema_path = 'schemas/{id}'.format(**schema_obj)
-    schema_id = schema_obj.get('id')
+    schema_id = mso.get_schema_details(schema)
+    schema_path = "schemas/{0}".format(schema_id)
```
Comment (lhercot) [inline plugins/modules/mso_schema_site_anp_epg_staticport.py:288]: should we just return this as part of the previous method as that path should always be the same.

**Review by lhercot — CHANGES_REQUESTED** [review]

**Comment by yannickcastano** [comment]
Hi team,
Other modules are also using this #Get schema_id portion of code (probably all the mso_schema_*.py). Can you find all the module using it in order to apply the optimisation ?
    # Get schema_id
    schema_obj = mso.get_obj('schemas', displayName=schema)
    if not schema_obj:
        mso.fail_json(msg="Provided schema '{0}' does not exist".format(schema))

    schema_path = 'schemas/{id}'.format(**schema_obj)
    schema_id = schema_obj.get('id')
Thanks,
/Yannick

**Comment by lhercot** [comment]
@yannickcastano Yes, this is just the first review. Wanted to check it before we apply it across the board.

**Pull request closed without merging**